frontend/src/register.tsx

```diff
@@ -0,0 +1,135 @@
+import {Bounce, toast, ToastContainer} from "react-toastify";
+import 'react-toastify/dist/ReactToastify.css';
+import React from "react";
+import {useNavigate} from "react-router";
+
+
+// Define the interface for the request payload
+interface UserPayload {
+    username: string;
+    email: string;
+    firstName: string;
+    lastName: string;
+}
+
+function Register() {
+
+  const navigate = useNavigate();
+// Function to create a new user
+    async function createUser(payload: UserPayload): Promise<void> {
+        const url = 'http://localhost:5000/users/new';
+
+        try {
+            const response = await fetch(url, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify(payload),
+            });
+
+            if (!response.ok) {
+                // Handle response errors
+                const errorResponse = await response.json()
+                toast.error(errorResponse.error);
+                console.error(response)
+                return
+                // throw new Error(`Error: ${response.status} - ${errorRes}`);
+            }
+
+            const data = await response.json();
+            console.log('User created successfully:', data);
+            toast.success("Success! Redirecting home !");
+
+            setTimeout(() => {
+                navigate("/");
+            }, 3000)
```

Comment:
Would it be possible to bind this to the toast somehow, so the navigation would be triggered once the toast disappears? A random `setTimeout` looks hacky to me.